src/core/lib/view-state.js

```diff
@@ -16,6 +16,7 @@ const defaultProps = {
 export default class ViewState {
 
   constructor(props, constraints, userData = {}) {
+    props = this._translateOrbitProps(props);
```

Comment:
Why do we need special handling for orbit viewport props? Will we need to write special handling for every viewport in the future? Can users not implement their own viewport classes?

Reply:
> Why do we need special handling for orbit viewport props? Will we need to write special handling for every viewport in the future? Can users not implement their own viewport classes?

I'd like the team's input on this. My preference would be that this is not build into viewport each viewport that really wants to offer a separate set of parameters as a "view" of the viewState offers functions to do the conversions.

```js
OrbitController.getOrbitParametersFromViewState(viewState)
OrbitController.getViewStateFromOrbitParameters(orbitParameters)
```

I'd rather prefer not to have special controller classes add new viewstate classes unless they need to opt out of this system completely (maybe because they try to implement some special projection system etc).

The reason is that those classes add nothing new except some different parameter names but they complicate the overall system.